editor/editor/layout.js

```diff
@@ -17,8 +17,7 @@ class Layout extends wp.element.Component {
 	}
 
 	switchMode( newMode ) {
-		// TODO: we need a serializer from blocks here
-		const html = this.state.html;
+		const html = this.mode === 'text' ? this.state.html : wp.blocks.serialize( this.state.blocks );
```

Comment:
Also, can we not store `html` and `blocks` in state and instead perform this logic in `render?